satpy/readers/osisaf_l3_nc.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""A reader for OSI-SAF level 3 products in netCDF format."""
+
+import logging
+from datetime import datetime
+
+import numpy as np
+
+from satpy.readers.netcdf_utils import NetCDF4FileHandler
+
+logger = logging.getLogger(__name__)
+
+
+class OSISAFL3NCFileHandler(NetCDF4FileHandler):
+    """Reader for the OSISAF l3 netCDF format."""
+
+    @staticmethod
+    def _parse_datetime(datestr):
+        try:
+            return datetime.strptime(datestr, "%Y-%m-%d %H:%M:%S")
+        except ValueError:
+            try:
+                return datetime.strptime(datestr, "%Y%m%dT%H%M%SZ")
+            except ValueError:
+                return datetime.strptime(datestr, "%Y-%m-%dT%H:%M:%SZ")
```

Comment:
Can this be moved closer to where it is used?

You could also do:

```python
for dt_format in ("%Y...", ...):
    try:
        return datetime.strptime(datestr, dt_fmt)
    except ValueError:
        continue
raise ValueError(f"Unsupported date format: {datestr}")

Reply:
Fixed!